drivers/storage/portworx/component/telemetry.go

```diff
@@ -1004,41 +1008,59 @@ func getArcusRegisterProxyURL(cluster *corev1.StorageCluster) string {
 }
 
 // CanAccessArcusRegisterEndpoint checks if telemetry registration endpoint is reachable
-func CanAccessArcusRegisterEndpoint(cluster *corev1.StorageCluster) (bool, error) {
+// return true immediately if it can reach to Arcus
+// return false after failing 5 times in a row
+func CanAccessArcusRegisterEndpoint(
+	cluster *corev1.StorageCluster,
+	httpProxy string,
+) bool {
 	endpoint := getArcusRegisterProxyURL(cluster)
 	logrus.Debugf("checking whether telemetry registration endpoint %s is accessible on cluster %s",
 		endpoint, cluster.Name)
 
-	url, err := url.Parse(fmt.Sprintf("https://%s:443/auth/1.0/ping", endpoint))
-	if err != nil {
-		return false, err
-	}
-
+	url, _ := url.Parse(fmt.Sprintf("https://%s:443/auth/1.0/ping", endpoint))
 	request := &http.Request{
 		Method: "GET",
 		URL:    url,
 		Header: map[string][]string{
 			"product-name": {"portworx"},
-			"appliance-id": {cluster.Status.ClusterUID},
+			"appliance-id": {"portworx"},
 			"component-sn": {cluster.Name},
 		},
 	}
 	client := &http.Client{}
-	response, err := client.Do(request)
-	if err != nil {
-		logrus.WithError(err).Warnf("failed to access telemetry registration endpoint %s", endpoint)
-		return false, nil
-	} else if response.StatusCode != 200 {
-		defer response.Body.Close()
-		body, _ := io.ReadAll(response.Body)
-		logrus.WithFields(logrus.Fields{
-			"code": response.StatusCode,
-			"body": string(body),
-		}).Warnf("failed to access telemetry registration endpoint %s", endpoint)
-		return false, nil
+	if httpProxy != "" {
+		if !strings.HasPrefix(strings.ToLower(httpProxy), "http://") {
+			httpProxy = "http://" + httpProxy
+		}
+		proxyURL, _ := url.Parse(httpProxy)
+		client.Transport = &http.Transport{
+			Proxy: http.ProxyURL(proxyURL),
+		}
 	}
-
-	return true, nil
+	for i := 1; i <= arcusPingRetry; i++ {
+		response, err := client.Do(request)
+		warnMsg := fmt.Sprintf("failed to access telemetry registration endpoint %s", endpoint)
+		if err != nil {
+			logrus.WithError(err).Warnf(warnMsg)
+		} else if response.StatusCode != 200 {
+			body, _ := io.ReadAll(response.Body)
+			response.Body.Close()
+			logrus.WithFields(logrus.Fields{
+				"code": response.StatusCode,
+				"body": string(body),
```

Comment:
this means it's accessible, we don't need to retry on this. I think it should return true immediately.

Reply:
we discussed before to send the request properly, and only consider it's a successful ping when got 200 OK response.

Reply:
Right, i remember that, do you know when it will not return 200? technically we don't care if it's 200, as long as the network is accessible